src/envoy/http/error_translator/filter.cc

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/envoy/http/error_translator/filter.h"
+
+#include <string>
+
+#include "common/http/headers.h"
+#include "external/envoy/include/envoy/stream_info/_virtual_includes/stream_info_interface/envoy/stream_info/stream_info.h"
+#include "src/envoy/utils/filter_state_utils.h"
+
+namespace espv2 {
+namespace envoy {
+namespace http_filters {
+namespace error_translator {
+
+using ::Envoy::Http::FilterHeadersStatus;
+
+Filter::Filter(FilterConfigSharedPtr config) : config_(config) {}
+
+// TODO(nareddyt): encodeHeaders or encodeData?
+// Probably need both for `handleEnvoyFilterError`
+FilterHeadersStatus Filter::encodeHeaders(Envoy::Http::ResponseHeaderMap&,
```

Comment:
This is tricky,  we need to study more envoy code here.  

In general,  in encodeHeader we should
if there is our rpcStatus, we should Stop, otherwise continue

in encodeData() or encodeTrailer() we should 
* generate our JSON body, and change headers content-type and content-length and continue

But not sure if this works for the data from "sendLocalReply",  it should

Reply:
We need to skip is_grpc:

Just read this [code](https://cs.corp.google.com/piper///depot/google3/third_party/envoy/src/source/common/http/utility.cc?rcl=307319856&g=0&l=435):

We should not write response body to grpc request.  

we can use content-type to detect grpc.

Reply:
Done as described above.